### ISSUE TITLE
Problem with re-exporting modules and node-style paths

### ISSUE THREAD

**Issue by zgredzik** [fuse-box/fuse-box#46]
When using node-style module paths (by dropping the `index.(j|t)sx?` suffix) fuse-box seems not to bundle the entire project.

It is worth noting that even when explicitly targeting the index files **with** the extension the project still won't bundle correctly, as the `.tsx` files won't get bundled unless they are also provided with the extension.

Unfortunately that causes the ts compiler to fail (this does not affect the bundling process ofc), as it does not accept extensions in import paths:
`src/app.tsx(4,41): error TS2691: An import path cannot end with a '.ts' extension. Consider importing './components/index' instead.`

My first guess regarding at least part of the issue would be `.tsx` files not triggering the `tsMode`, as mentioned in #45 ?

Example: https://github.com/zgredzik/fusebox-reexport-test

fuse-box logs:
```php
└── default (3 files,  614 Bytes)
      app.tsx
      components.js <--
      components/componentA.js <--

    --------------
    Size: 614 Bytes
    Time: 329ms
    --------------
```

logs with explicitly targetting `index.ts` (https://github.com/zgredzik/fusebox-reexport-test/tree/fixed-index-paths):
```php
└── default (7 files,  1.8 kB)
      app.tsx
      components/index.ts
      components/componentA/index.ts
      components/componentA/componentA.js <--
      components/componentA/componentA2.js <--
      components/componentB/index.ts
      components/componentB/componentB.js <--

    --------------
    Size: 1.8 kB
    Time: 344ms
    --------------
```

logs with extensions provided in all import paths (https://github.com/zgredzik/fusebox-reexport-test/tree/fixed-all-paths):
```php
└── default (7 files,  4.6 kB)
      app.tsx
      components/index.ts
      components/componentA/index.ts
      components/componentA/componentA.tsx
      components/componentA/componentA2.tsx
      components/componentB/index.ts
      components/componentB/componentB.tsx

    --------------
    Size: 4.6 kB
    Time: 415ms
    --------------
```

**Comment by nchanged** [comment]
Thanks! I think it's clear enough, we will get to it soon.

**Comment by nchanged** [comment]
It should be fixed in `1.3.34` please, test all possible scenarios.
